File: examples/go/security_policy.go
// Example: Use whitelist/blacklist security policies for bash and curl.
package main

import (
	"context"
	"fmt"
	"os"

	"github.com/steven0lisa/agent-libs/go"
)

func main() {
	cfg := agentlib.DefaultConfig()
	cfg.APIKey = os.Getenv("ANTHROPIC_AUTH_TOKEN")

	// Bash: allow git and ls; block rm.
	cfg.BashWhitelist = []agentlib.Pattern{
		{Pattern: "git *", Type: "wildcard"},
		{Pattern: "ls *", Type: "wildcard"},
	}
	cfg.BashBlacklist = []agentlib.Pattern{
		{Pattern: "rm *", Type: "wildcard"},
		{Pattern: "^dd\\s", Type: "regex"},
	}

	// Curl: allow example.com; block known bad domains.
	cfg.CurlWhitelist = []agentlib.Pattern{
		{Pattern: "*.example.com/*", Type: "wildcard"},
	}
	cfg.CurlBlacklist = []agentlib.Pattern{
		{Pattern: "evil\\.com|malicious\\.org", Type: "regex"},
	}

	agent := agentlib.NewAgent(cfg)
	ctx := context.Background()

	for event := range agent.Run(ctx, "Show me the git status of this repository") {
		switch event.Type {
		case agentlib.EventMessageDelta:
			fmt.Print(event.Data["text"])
		case agentlib.EventComplete:
			fmt.Printf("\n\n[Complete] %s\n", event.Data["final_content"])
		case agentlib.EventError:
			fmt.Printf("\n[Error] %s\n", event.Data["message"])
		}
	}
}

File: examples/rust/security_policy.rs
// Example: Use whitelist/blacklist security policies.
use agentlib::{Agent, AgentConfig, Pattern};
use std::env;

#[tokio::main]
async fn main() {
    let config = AgentConfig {
        api_key: env::var("ANTHROPIC_AUTH_TOKEN").unwrap_or_default(),
        bash_whitelist: vec![
            Pattern::new("git *", agentlib::PatternType::Wildcard),
            Pattern::new("ls *", agentlib::PatternType::Wildcard),
        ],
        bash_blacklist: vec![
            Pattern::new("rm *", agentlib::PatternType::Wildcard),
            Pattern::new(r"^dd\s", agentlib::PatternType::Regex),
        ],
        curl_whitelist: vec![
            Pattern::new("*.example.com/*", agentlib::PatternType::Wildcard),
        ],
        curl_blacklist: vec![
            Pattern::new(r"evil\.com|malicious\.org", agentlib::PatternType::Regex),
        ],
        ..Default::default()
    };

    let mut agent = Agent::new(config);
    let mut rx = agent.run("Show me the git status of this repository").await.unwrap();

    while let Some(event) = rx.recv().await {
        match event {
            agentlib::Event::MessageDelta { text } => print!("{}", text),
            agentlib::Event::Complete { final_content } => println!("\n\n[Complete] {}", final_content),
            agentlib::Event::Error { message } => println!("\n[Error] {}", message),
            _ => {}
        }
    }
}
